src/crs_linter/linter.py

```diff
@@ -366,99 +426,133 @@ def check_tx_variable(self):
-                        val_act = re.findall(r"%\{(tx.[^%]*)\}", a['act_arg'], re.I)
-                    if "act_arg_val" in a and a['act_arg_val'] is not None:
-                        val_act_arg = re.findall(r"%\{(tx.[^%]*)\}", a['act_arg_val'], re.I)
+                    if "act_arg" in a and a["act_arg"] is not None:
+                        val_act = re.findall(r"%\{(tx.[^%]*)\}", a["act_arg"], re.I)
+                    if "act_arg_val" in a and a["act_arg_val"] is not None:
+                        val_act_arg = re.findall(
+                            r"%\{(tx.[^%]*)\}", a["act_arg_val"], re.I
+                        )
                     for v in val_act + val_act_arg:
                         v = v.lower().replace("tx.", "")
                         # check whether the variable is a captured var, eg TX.1 - we do not care that case
                         if not re.match(r"^\d$", v, re.I):
                             # v holds the tx.ANY variable, but not the captured ones
                             # we should collect these variables
-                            if (v not in self.globtxvars or phase < self.globtxvars[v]['phase']):
-                                self.undef_txvars.append({
-                                    'var': v,
-                                    'ruleid': ruleid,
-                                    'line': a['lineno'],
-                                    'endLine': a['lineno'],
-                                    'message': "TX variable '%s' not set / later set (rvar) in rule %d" % (v, ruleid)
-                                })
+                            if (
+                                v not in self.globtxvars
+                                or phase < self.globtxvars[v]["phase"]
+                            ):
+                                self.error_undefined_txvars.append(
+                                    {
+                                        "var": v,
+                                        "ruleid": ruleid,
+                                        "line": a["lineno"],
+                                        "endLine": a["lineno"],
+                                        "message": f"TX variable '{v}' not set / later set (rvar) in rule {ruleid}",
+                                    }
+                                )
                             else:
-                                self.globtxvars[v]['used'] = True
+                                self.globtxvars[v]["used"] = True
                         else:
                             if v in self.globtxvars:
-                                self.globtxvars[v]['used'] = True
-                    aidx += 1
+                                self.globtxvars[v]["used"] = True
 
                 if "operator_argument" in d:
-                    oparg = re.findall(r"%\{(tx.[^%]*)\}", d['operator_argument'], re.I)
+                    oparg = re.findall(r"%\{(tx.[^%]*)}", d["operator_argument"], re.I)
```

Comment:
Here you removed the `\` from the front of the close bracket. Is that accidentally or on purpose?

Reply:
In Python regular expressions, special characters that need to be escaped outside of character classes include: *, +, ?, {, (, ), ^, $, |, and \. Inside character classes, you need to escape ], -, ^, and \ if they are used in a way that could confuse the regex parser.

Reply:
> In any case, formatting should be consistent.

Yeah, this is what I noticed.